File: 4_algorithm_python/1_Two Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen = {}
        for i in range(0, len(nums)):
            diff = target - nums[i]
            if diff in seen.keys():
                return [seen[diff], i]
            else:
                seen[nums[i]] = i

File: 4_algorithm_python/1_Two Sum.java
import java.util.ArrayList;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<Integer, Integer>();
        int[] list = new int[2];
        for (int i = 0; i < nums.length; i++) {
            int diff = target - nums[i];
            if (map.containsKey(diff)) {
                list[0] = map.get(diff);
                list[1] = i;
                break;
            } else
                map.put(nums[i], i);
        }
        return list;
    }
}